presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -145,13 +150,21 @@ public PlanNode visitTableScan(TableScanNode node, RewriteContext<Void> context)
                     originalConstraint);
         }
 
+        @Override
+        public PlanNode visitGroupId(GroupIdNode node, RewriteContext<Void> context)
+        {
+            isGroupIdNodePresent = true;
```

Comment:
Shouldn't 'isGroupIdNodePresent' flag be cleared in projection above grouping? E.g consider subqueries in plan:
`select ... from (select ... from (select ..) group by ...) group by ...)`